### PR TITLE
[F] Metis: Add schema (#6)

### LINKED ISSUE #5: Metis: add schema for payload
Add schema for payload and check is correct.

See https://github.com/repositive/metis/pull/4#issuecomment-326529491

### PR DESCRIPTION
**Pull request by DrWilliamsSteven**
@Istar-Eldritch 

This PR:
* adds a JSON schema to check the payload
* updates the README

Closes #5 and closes repositive/agora#112.

### CODE REVIEW

File: src/annotate.ts [hunk 1]
```diff
@@ -1,9 +1,20 @@
 import * as _request from 'request-promise';
+import * as Ajv from 'ajv';
+import * as fs from 'fs';
 
 //-----------------------
 
 export default async function annotate(opts: { payload: { field: string, term: string } }) {
 
+  const schema: any = JSON.parse(fs.readFileSync('./schemas/is-valid.json', 'utf8'));
```
Comment (Istar-Eldritch) [inline src/annotate.ts]: I know is a WIP but remember you should move the loading of the schema outside of the annotate function, mainly when when the loading function `readFileSync` is blocking the thread.

File: src/annotate.spec.ts [hunk 2]
```diff
@@ -133,6 +132,7 @@ test('Testing Zooma annotate service', (t: Test) => {
         console.error(err);
         console.log('annotate error');
         // API call failed...
+        st.notOk(err, 'There is an error');
```
Comment (DrWilliamsSteven) [inline src/annotate.spec.ts]: @Istar-Eldritch I wanted to check that an error is thrown, but this is the only way I could get it to work. Is there a better way?

File: src/annotate.spec.ts [hunk 3]
```diff
@@ -97,12 +99,9 @@ test('Testing Zooma annotate service', (t: Test) => {
 
     const result = await _annotate.default({ payload: { field, term } })
       .then(function(data: any) {
-        //console.log('got data', data);
         return data;
       }).catch((err: any) => {
-        console.error(err);
-        console.log('annotate error');
-        // API call failed...
+        st.ok(err, 'There is an error');
```
Comment (DrWilliamsSteven) [inline src/annotate.spec.ts]: @Istar-Eldritch I wanted to check that an error is thrown, but this is the only way I could get it to work. Is there a better way?

Comment (Istar-Eldritch) [reply src/annotate.spec.ts]: Yep, this is correct. :)

Comment (DrWilliamsSteven) [reply src/annotate.spec.ts]: OK thanks.
Let me know if there's anything else I need for this PR. I think it should be complete now.

Comment (DrWilliamsSteven) [reply src/annotate.spec.ts]: Also, if you are able to set up/import the environment variables for metis on CircleCI that would be great.

Comment (Istar-Eldritch) [reply src/annotate.spec.ts]: Sure. Sorry about the CI thing, we need to check why you don't have permissions on it.

Comment (DrWilliamsSteven) [reply src/annotate.spec.ts]: I think you need to be a github admin

Comment (DrWilliamsSteven) [reply src/annotate.spec.ts]: On the import variables button for me it says:
"To use this function you must be an organisation admin for your VCS-provider"

Comment (Istar-Eldritch) [reply src/annotate.spec.ts]: That would make sense. I just added the env vars. I would like to review this properly one more time wefore giving green light ;)

**Pull request merged**